FlowCal/plot.py

```diff
@@ -64,6 +67,511 @@
+
+    Notes
+    -----
+    Logicle scaling combines the advantages of logarithmic and linear
+    scaling. It is useful when data spans several orders of magnitude
+    (when logarithmic scaling would be appropriate) and a significant
+    number of datapoints are negative.
+
+    Logicle scaling is implemented using the following equation::
+
+        x = T * 10**(-(M-W)) * (10**(s-W) \
+                - (p**2)*10**(-(s-W)/p) + p**2 - 1)
+
+    This equation transforms data ``s`` expressed in "display scale" units
+    into ``x`` in "data value" units. Parameters in this equation
+    correspond to the class properties. ``p`` and ``W`` are related as
+    follows::
+
+        W = 2*p * log10(p) / (p + 1)
+
+    If a FCSData object or list of FCSData objects is specified along with
+    a channel, the following default logicle parameters are used: T is
+    taken from the largest ``data[i].range(channel)[1]`` or the largest
+    element in ``data[i]`` if ``data[i].range()`` is not available, M is
+    set to ``4.5 / np.log10(262144) * np.log10(T)``, and W is taken from
+    ``(M - log10(T / abs(r))) / 2``, where ``r`` is the minimum negative
+    event. If no negative events are present, W is set to zero.
+
+    References
+    ----------
+    .. [1] D.R. Parks, M. Roederer, W.A. Moore, "A New Logicle Display
+    Method Avoids Deceptive Effects of Logarithmic Scaling for Low Signals
+    and Compensated Data," Cytometry Part A 69A:541-551, 2006, PMID
+    16604519.
+
+    """
+    # ``input_dims``, ``output_dims``, and ``is_separable`` are required by
+    # matplotlib.
+    input_dims = 1
+    output_dims = 1
+    is_separable = True
+    # Locator objects need this object to store the logarithm base used as an
+    # attribute.
+    base = 10
+
+    def __init__(self, T=None, M=None, W=None, data=None, channel=None):
+        matplotlib.transforms.Transform.__init__(self)
+        # If data is included, try to obtain T, M and W from it
+        if data is not None:
+            if channel is None:
+                raise ValueError("if data is provided, a channel should be"
+                    + " specified")
+            # Convert to list if necessary
+            if not isinstance(data, list):
+                data = [data]
+            # Obtain T, M, and W if not specified
+            # If elements of data have ``.range()``, use it to determine the
+            # max data value. Else, use the maximum value in the array.
+            if T is None:
+                T = 0
+                for d in data:
+                    # Extract channel
+                    y = d[:, channel] if d.ndim > 1 else d
+                    if hasattr(y, 'range') and hasattr(y.range, '__call__'):
+                        Ti = y.range(0)[1]
+                    else:
+                        Ti = np.max(y)
+                    T = Ti if Ti > T else T
+            if M is None:
+                M = 4.5 / np.log10(262144) * np.log10(T)
```

Comment:
you originally had `M = 4.5` (which I think came straight from the Logicle paper?), but added this scaling factor. `np.log10(262144)` also comes from the paper? and you've just made this scale to the data a little better by effectively incorporating a ratio of `T` (extracted as max value from data?) and the max value used in the paper (262144)? is this right?